### ISSUE TITLE
v1.0.0: Improved behaviour when remote is not available

### ISSUE THREAD

**Issue by ReCREWt-Fabi** [module-federation/vite#76]
Hi there!

First of all thank you @zhangHongEn for your work!

We've previously used @originjs/vite-plugin-federation but decided to look for an alternative due to it being seemingly abandoned.
I've migrated our microfrontends to this plugin, and so far almost everything is working fine. The only problem occurs when a remote module is not available e.g. during development. 

With the old plugin an unreachable remote did not cause an error until actually trying to import the component. Currently this is not the case in this plugin, which would require us to always start the remote frontends locally during development and could cause a complete crash, if a single remote is unavailable when deployed to our server.

Could you add some kind of fallback mechanism or another way to make the plugin more tolerant against missing remotes? 

**Current behaviour**
Remote is unreachable -> App crashes during initialization -> blank screen

![image](https://github.com/user-attachments/assets/3910e0f0-60aa-4d14-902b-2d9c1fb93b63)

**Desired behaviour**
App initializes correctly and only raises an error, when actually trying to import the exposed remote component

**My host config**
```
...
        federation({
            name: "host",
            filename: "remoteEntry.js",
            remotes: {
                mfe1: {
                    entry: "http://localhost:3004/remoteEntry.js",
                    type: "module"
                },
                mfe2: {
                    entry: "http://localhost:3005/remoteEntry.js",
                    type: "module"
                }
            },
            shared
        }),
...
```

**Comment by zhangHongEn** [comment]
OK, I will optimize it so that it won't be registered if it's not used.

**Comment by zhangHongEn** [comment]
But we have to wait until the SSR function is completed before optimizing it

**Comment by gioboa** [comment]
Closed by #86
